org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/IProjectImporter.java

```diff
@@ -24,6 +26,13 @@
 
 	boolean applies(IProgressMonitor monitor) throws OperationCanceledException, CoreException;
 
+	/**
+	 * Check whether the importer applies to the given project configurations.
+	 * @param projectConfigurations Collection of the project configurations.
+	 * @param monitor progress monitor.
+	 */
+	boolean applies(Collection<IPath> projectConfigurations, IProgressMonitor monitor);
```

Comment:
I would add a default implementation for it to avoid breaking importers from other side Java plugins. If so, you need to remove the override method in `AbstractProjectImporter.java `.

```java
default boolean applies(Collection<IPath> projectConfigurations, IProgressMonitor monitor) {
    return applies(monitor);
}
```

Reply:
Updated